### PR TITLE
Filename Collisions with downloadable products

### LINKED ISSUE #13915: Filename Collisions with downloadable products
### Preconditions

1. Magento 2.4
### Steps to reproduce

1. Create a downloadable product and add a downloadable link and upload a file for the link with a filename on the local system like 'book.pdf'
2. Create a different downloadable produdt and add a downloadable link and upload a different file with the same name on the local system like 'book.pdf'
3. Each book.pdf is a different file with different contents possibly located on different places on the users computer when it is uploaded but has the same filename.

### Expected result

1. Each downloadable product will have a different file associated with it.

### Actual result

1. The last uploaded file replaces the existing file. So both downloadable products are associated with the same file. 

The problem is the uploader does not check to see if there is an existing file at the location it is trying to upload a file to but overrides it making the unexpected result. The media uploader doesn't seem to have this problem only downloadable product links and probably samples too.

### PR DESCRIPTION
**Pull request by engcom-Charlie**
### Description (*)

Fixed and covered by test.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#13915

### Manual testing scenarios (*)
See #13915

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Review by eduard13 — CHANGES_REQUESTED** [review]
Hi @engcom-Charlie, thank you for the fix. 
Looks like there are couple of constantly failing [DB tests](https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/28726/36eed8f64eaebfe44841fd91c8661f11/DatabaseCompare/ce-skip-sd-db-compare-report.html) (all Sample Data checks).

Additionally, as a suggestion, may we have more meaningful commit messages?
Thank you.

File: app/code/Magento/Downloadable/Helper/File.php [hunk 1]
```diff
@@ -6,52 +6,64 @@
 namespace Magento\Downloadable\Helper;
 
 use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\App\Helper\AbstractHelper;
+use Magento\Framework\App\Helper\Context;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Filesystem;
+use Magento\Framework\Filesystem\Directory\WriteInterface;
+use Magento\Framework\Filesystem\Driver\File as FileDriver;
```
Comment (Nazar65) [inline app/code/Magento/Downloadable/Helper/File.php:14]: Please use interface for that Magento\Framework\Filesystem\DriverInterface; and inject Magento\Framework\Filesystem\Driver\File via di configuration

**Review by Nazar65 — CHANGES_REQUESTED** [review]

**Comment by sidolov** [comment]
@engcom-Charlie  I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**